taichi/program/field_info.h

```diff
@@ -29,6 +30,7 @@ struct FieldInfo {
   DEFINE_PROPERTY(int, matrix_rows);
   DEFINE_PROPERTY(int, matrix_cols);
   DEFINE_PROPERTY(std::vector<int>, shape);
+  DEFINE_PROPERTY(uint64_t, num_elements);
   DEFINE_PROPERTY(FieldSource, field_source);
```

Comment:
Out of curiosity, when we get rid of the interops, can we remove all field stuff in GGUI? I'd assume they are converted to ndarrays first?

Reply:
That's the plan. FieldInfo will become basically a wrapped DeviceAllocation